doschema/validation.py

```diff
@@ -0,0 +1,165 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of DoSchema.
+# Copyright (C) 2016 CERN.
+#
+# DoSchema is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# DoSchema is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with DoSchema; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Merger module."""
+
+import collections
+
+from doschema.errors import DoSchemaException, JSONSchemaCompatibilityException
+
+
+def start(schemas_list, ignore_index=True, fields_types_dict={}):
+    """Start processing a given list of schemas.
+
+    :param schemas_list: List of schemas to check compatibility.
+    :param ignore_index: If set to True, which is default, it will ignore
+                         array indexes.
+    """
+    fields_types_dict = fields_types_dict or {}
+    ind = 0
+    for scheme in schemas_list:
+        _JSONSchemaCompatibilityChecker(
+            ind, (), scheme, ignore_index
+        ).traverse(fields_types_dict)
+        ind += 1
+
+    return fields_types_dict
+
+
+FieldToAdd = collections.namedtuple(
+    'FieldToAdd',
+    'schema_index field_tuple field_type'
+)
+
+
+class _JSONSchemaCompatibilityChecker(object):
```

Comment:
If I am not mistaken this is an old file. If you want to keep it, it is better to create a local branch in your repository. Keeping it in the PR makes it harder to read.